communication/src/coap_channel.h

```diff
@@ -222,6 +234,12 @@ class __attribute__((packed)) CoAPMessage
 		CoAPType::Enum coapType = CoAP::type(get_data());
 		if (coapType==CoAPType::CON) {
 			timeout = now + transmit_timeout(transmit_count);
+			if (transmit_count == 0) {
+				safeIntegerDiagnosticIncrement(g_trasmittedMessageCounter);
```

Comment:
I tested with int max `g_trasmittedMessageCounter = 2147483647;` , and it wraps around to `0`